micrometer-core/src/main/java/io/micrometer/core/instrument/Observer.java

```diff
@@ -0,0 +1,29 @@
+/**
+ * Copyright 2017 Pivotal Software, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.micrometer.core.instrument;
+
+/**
+ * @author Dmitry Poluyanov
+ * @since 23.07.17
+ */
+public interface Observer {
```

Comment:
Is the `Observer` intended as an abstraction for a variety of summary statistics commonly found on timers and distribution summaries?

If not, read no further...

If so, note that we currently have two things you could call a summary stat: client side quantiles and histograms. 

Client-side quantiles are errorless (because of course Micrometer sees all the samples), but are not aggregable in general. For this reason, I believe they are of little value in systems of sufficient scale. By sufficient scale, I mean when the number of instances (or containers) reporting a metric is large enough that grouping the metric by instance results in an impractical number of lines to plot on a graph and still be able to visualize an appreciable difference between instances.

Histograms really are a discretization of the continuous input domain whose chief purpose is to compute an aggregable quantile approximation through server-side math in the monitoring system. Their weakness lies in the essentially unbounded potential of their error bound, depending on how well (or poorly) the buckets are chosen. And the bucket choosing problem is unfortunately non-trivial.

If there was a way for us to infer a good bucket selection that would yield an acceptable error bound, I would be very tempted to strike client-side quantile support entirely (or at least move it out of timers and distribution summaries as a toy that you can record separately). I just need a little more time to research this. I believe Netflix has some recent research on a fixed bucket set that empirically works well for a wide variety of timers.

There are other summary stats I expect we'll hear requests for at some point, e.g. specifically client-side standard deviation. Many of them assume a normal distribution, which experience has shown is unfortunately rare.

All this is a super long winded way of saying: we may not ultimately need an abstraction if the only summary stat is a histogram?